### DISCUSSION TITLE
How to get crack location

### DISCUSSION THREAD

**Discussion by xingwang01** [idaholab/moose#26379] [Q&A Modules: Porous Flow]
I use BreakMeshByBlockGenerator and CohesiveZoneMaster for simulating crack propagation, and I want to get the location of the crack in the process of crack propagation and crack width information, such as used within the fracture fluid is calculated. How can I get this information.

**Comment by GiudGiud** [comment]
@recuero

**Reply by GiudGiud** [reply]
@hugary1995

**Reply by jiangwen84** [reply]
Are you planning to get those information in the code or just visualize them in the paraview?

The crack opening displacement and damage parameter (indicate the crack locations) are are stored as material properties, so you should be able to get those through getMaterialProperties API or look at the output Aux Variable in Paraview.

**Reply by xingwang01** [reply]
I want to use the cohesive force model to simulate the fracture propagation under fluid-structure coupling, but the model either does not propagate or does not converge.  I would like to know if the fluid field was taken into account when balancing forces within the crack. If fluid mechanics is considered, how to solve convergence, if not how to deal with the fluid mechanics balance in the fracture.
